pkg/operation/botanist/component/logging/fluentoperator/fluent_operator.go

```diff
@@ -0,0 +1,501 @@
+	OperatorPriorityClassName string
+	// FluentBitPriorityClass is the name of the priority class of the Fluent-bit.
+	FBPriorityClass string
+}
+type fluentOperator struct {
+	client            client.Client
+	namespace         string
+	values            Values
+	additionalInputs  []*fluentresources.ClusterInput
+	additionalFilters []*fluentresources.ClusterFilter
+	additionalParsers []*fluentresources.ClusterParser
+}
+
+// New creates a new instance of fluent operator.
+func New(
+	client client.Client,
+	namespace string,
+	values Values,
+	additionalInputs []*fluentresources.ClusterInput,
+	additionalFilters []*fluentresources.ClusterFilter,
+	additionalParsers []*fluentresources.ClusterParser,
+) Interface {
+	return &fluentOperator{
+		client:            client,
+		namespace:         namespace,
+		values:            values,
+		additionalInputs:  additionalInputs,
+		additionalFilters: additionalFilters,
+		additionalParsers: additionalParsers,
+	}
+}
+
+func (f *fluentOperator) Deploy(ctx context.Context) error {
+	var (
+		operatorRegistry                = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+		operatorCustomResourcesRegistry = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+
+		serviceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: f.namespace,
+				Labels:    f.getLabels(),
+			},
+			AutomountServiceAccountToken: pointer.Bool(false),
+		}
+		clusterRole = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   roleName,
+				Labels: f.getLabels(),
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"daemonsets", "statefulsets"},
+					Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods"},
+					Verbs:     []string{"get"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"secrets", "configmaps", "serviceaccounts", "services", "namespaces"},
+					Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+				},
+				{
+					APIGroups: []string{"fluentbit.fluent.io"},
+					Resources: []string{"fluentbits", "clusterfluentbitconfigs", "clusterfilters", "clusterinputs", "clusteroutputs", "clusterparsers"},
+					Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+				},
+				{
+					APIGroups: []string{"fluentd.fluent.io"},
+					Resources: []string{"fluentds", "fluentds/status", "clusterfluentdconfigs", "clusterfluentdconfigs/status", "fluentdconfigs", "fluentdconfigs/status", "clusterfilters", "filters", "clusteroutputs", "outputs"},
+					Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+				},
```

Comment:
Currently, the operator has no option to disable the `fluend` controller and removing these roles from it will cause errors in the operator.
Ref: https://github.com/fluent/fluent-operator/issues/612